Paintroid/src/main/java/org/catrobat/paintroid/tools/helper/FillAlgorithm.java

```diff
@@ -136,4 +139,13 @@ private void checkRangeAndGenerateNewRanges(Range range, int row, boolean direct
 		}
 	}
 
+	private boolean isPixelWithinColorTolerance(int pixel) {
```

Comment:
Since this method is called very often, I propose to make it a bit faster, as follows: Replace `mColorToleranceThreshold` by `mColorToleranceThresholdSquared` to avoid having to call the cpu intensive `Math.sqrt` method for each pixel. 

Additionally, to make it even more efficient, avoid the intermediate `diff` variable.

The code thus could look as follows:

	private boolean isPixelWithinColorTolerance(int pixel) {
		return ColorToleranceThresholdSquared >=
                       Math.pow(Color.red(pixel) - Color.red(mReplacementColor), 2)
			+ Math.pow(Color.green(pixel) - Color.green(mReplacementColor), 2)
			+ Math.pow(Color.blue(pixel) - Color.blue(mReplacementColor), 2)
			+ Math.pow(Color.alpha(pixel) - Color.alpha(mReplacementColor), 2);
	}

I suggest to add a comment that explains the reason for this speed optimization, so that it does not get undone during code review.

`mColorToleranceThresholdSquared` needs to be computed only once per fill command in line 58, as shown above.